### DISCUSSION TITLE
How does the asset_selection parameter for a sensor interact with RunRequests?

### DISCUSSION THREAD

**Discussion by dpeng817** [dagster-io/dagster#13959] [General]
You can specify an asset selection on a sensor, and also an asset selection on run requests. Does the asset selection specified on a run request operate on the already-defined asset subselection, or on the entire asset graph?

**Comment by dpeng817** [comment]
https://dagster.slack.com/archives/C01U954MEER/p1682609519848859

**Comment by sryza** [comment]
The asset selection specified on a run request operates on the already-defined asset subselection. I.e., if `@sensor` has the `asset_selection` parameter set, then it will never assets that are outside of that selection. If a `RunRequest` includes an `asset_selection` it will just sub-select the sensor-level selection.

We should eventually probably add this to the apidoc for the `asset_selection` parameter of `RunRequest`.

**Reply by cbini** [reply]
Thanks @sryza, that's exactly what I was wondering! Clarifying it in the docs would be helpful.

**Reply by Siddharth545** [reply]
PLease give one example, how to select an asset in runrequest sensor

**Reply by yuhan** [reply]
Here's the apidoc: https://docs.dagster.io/_apidocs/schedules-sensors#dagster.RunRequest.asset_selection

```python
from dagster import AssetKey, Definitions, RunRequest, asset, define_asset_job, sensor


@asset
def asset1():
    return [1, 2, 3]


@asset
def asset2():
    return [4, 5]


all_asset_job = define_asset_job(name="all_asset_job", selection="*")

@sensor(job=all_asset_job)
def my_sensor():
    # certain conditions are met
    if ...:
        # this will only kick off asset1
        yield RunRequest(asset_selection=[AssetKey("asset1")])


defs = Definitions(assets=[asset1, asset2], sensors=[my_sensor])
```
